packages/preset-mini/src/_rules/typography.ts

```diff
@@ -13,14 +13,21 @@ export const fonts: Rule<Theme>[] = [
     /^text-(.+)$/,
     ([, s = 'base'], { theme }) => {
       const [size, leading] = splitShorthand(s, 'length')
-      const sizePairs = toArray(theme.fontSize?.[size])
-      const lineHeight = leading ? handleLineHeight(leading, theme) : undefined
+      const sizePairs = toArray(theme.fontSize?.[size]) as [undefined] | [string] | [string, string | CSSObject] | [string, string, string]
```

Comment:
This type looks not great, can you help me improve it?

Reply:
Can we add `toArratReadonly` function instead of `toArray`? The function is very familiar to `toArray` just a different type, and it returns types exactly what you want. I am not sure we need to want it.


```ts
type Arrayify<T> = T extends readonly any[] ? T : readonly [T]
export function toArrayReadonly<T>(value: T) {
  return (Array.isArray(value) ? value : [value]) as Arrayify<T>
}
```

Reply:
Thank you so much.